packages/customize-widgets/src/components/sidebar-block-editor/use-sidebar-block-editor.js

```diff
@@ -172,6 +172,11 @@ export default function useSidebarBlockEditor( sidebar ) {
 					return blockToWidget( nextBlock );
 				} );
 
+				// Bail out updates if the updated widgets are the same.
+				if ( isShallowEqual( sidebar.getWidgets(), nextWidgets ) ) {
+					return prevBlocks;
+				}
```

Comment:
This is to prevent changes from being pushed to the history stack when switching from edit form to preview mode in Legacy Widgets. This is probably a bug in the Legacy Widgets block though and we should probably fix it there. @noisysocks, any ideas why it triggers a new change even though the instance is the same?